install_files/ansible-base/roles/common/tasks/from_fpf_repo_install_grsec.yml

```diff
@@ -20,3 +20,48 @@
   apt: pkg={{ grsec_package }} state=latest
   async: 300
   poll: 10
+
+- name: get grsec kernel string
+  shell: grep menuentry /boot/grub/grub.cfg | grep grsec | grep -v recovery | head -1 | cut -d "'" -f2
+  register: grsec_str
+
+- name: set grsec kernel as default for next boot
+  command: grub-reboot "Advanced options for Ubuntu>{{ grsec_str.stdout }}"
+
+- stat: path="/proc/sys/kernel/grsecurity/grsec_lock"
+  register: running_grsec
+
+- name: reboot into the grsec kernel
+  command: shutdown -r now "Rebooting into the grsec kernel..."
+  async: 0
+  poll: 0
+  ignore_errors: true
+  when: not running_grsec.stat.exists
+  tags:
+    - reboot
+
+- name: waiting for server to come back
+  local_action:
+    module: wait_for
+      host={{ ansible_ssh_host }}
+      port={{ ansible_ssh_port }}
+      delay=45
+      state=started
```

Comment:
Should also add the `when: not running_grsec.stat.exists` config to this task. If the task to reboot the server doesn't run because you are already booted into grsec then you do not need to wait for the server to come back up.